### ISSUE TITLE
Search get empty result after release another partition

### ISSUE THREAD

**Issue by cydrain** [milvus-io/milvus#7250]
#### Steps/Code to reproduce:
tests/python_client/testcases/test_partition_20.py::TestPartitionParams::test_partition_release
```
    @pytest.mark.tags(CaseLabel.L1)
    # @pytest.mark.parametrize("search_vectors", [cf.gen_vectors(1, ct.default_dim)])
    def test_partition_release(self):
        """
        target: verify release partition
        method: 1. create a collection and several partitions
                2. insert data into each partition
                3. flush and load the partitions
                4. release partition1
                5. release partition1 twice
        expected: 1. the released partition is released
                  2. the other partition is not released
        """
        # create collection
        collection_w = self.init_collection_wrap()

        # create two partitions
        partition_w1 = self.init_partition_wrap(collection_w)
        partition_w2 = self.init_partition_wrap(collection_w)

        # insert data to two partition
        partition_w1.insert(cf.gen_default_list_data())
        partition_w2.insert(cf.gen_default_list_data())

        # load two partitions
        partition_w1.load()
        partition_w2.load()

        # search two partitions
        search_vectors = cf.gen_vectors(1, ct.default_dim)
        res1, _ = partition_w1.search(data=search_vectors,
                                      anns_field=ct.default_float_vec_field_name,
                                      params={"nprobe": 32}, limit=1)
        res2, _ = partition_w2.search(data=search_vectors,
                                      anns_field=ct.default_float_vec_field_name,
                                      params={"nprobe": 32}, limit=1)
        assert len(res1) == 1 and len(res2) == 1

        # release the first partition
        partition_w1.release()

        # check result
        res1, _ = partition_w1.search(data=search_vectors,
                                      anns_field=ct.default_float_vec_field_name,
                                      params={"nprobe": 32}, limit=1,
                                      check_task=ct.CheckTasks.err_res,
                                      check_items={ct.err_code: 1, ct.err_msg: "partitions have been released"})
        res2, _ = partition_w2.search(data=search_vectors,
                                      anns_field=ct.default_float_vec_field_name,
                                      params={"nprobe": 32}, limit=1)
        assert len(res2) == 1
```

#### Expected result:
The second search for partition_w2 should return valid result

#### Actual results:
after release partition_w1, search for partition_w2 return empty result

**Comment by bigsheeper** [comment]
The first partition and the second partition use the same virtual channel in query node, which means they use the same `tSafe` (for a virtual there is only one `tSafe` in query node in query node). But when the first partition is released, the `tSafe` would be removed, and the service time check of the second partition's searching in `streaming` would not be passed, which caused the null search result.
In conclusion, when there are multiple partitions with the same `tSafe`, `tSafe` should not be released unless all the partitons have been released. I'v added a `tSafe` referece counter to fix this in the branch of refactoring query workflow (#7025 ). But this pr would not be merged into 2.0.0-RC-5, so I removed service time check in `streaming` to fix this issue temporarily.
